src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -1490,5 +1504,15 @@ public async ValueTask<FlushResult> WriteAsyncAwaited(Task initializeTask, ReadO
                 return await Output.FlushAsync(cancellationToken);
             }
         }
+
+        private static bool GetDisableHeaderReuseSwitch()
+        {
+            if (AppContext.TryGetSwitch(DisableHeaderReuseSwitch, out var disableHeaderReuse))
```

Comment:
Is there a switch to disable the similar changes to the target parsing?

Reply:
No. I saw it more as a paranoia setting https://github.com/aspnet/AspNetCore/pull/8374#discussion_r272666738 as the headers could contain the cookies/auth info (whereas you shouldn't be putting this in the path).

Also the alternate path for startline would be quite messy. Can add it though?

Reply:
Is it really that messy? It is a little bit of paranoia, but I think having a switch that disable all reuse of previous materialize strings would relieve some of that paranoia.

Reply:
When I was originally thinking about how to do it; I thought it would be two branches with the two forms of parsing.

However, can just add it as first term in the `if` tests; so no it wouldn't be messy.

Reply:
@benaadams we were discussion this and think it should be a regular option with, not an app context switch. It should just be a single option that can also be read from configuration (like we do for a bunch of other settings in kestrel)

Reply:
Added config rather than switch https://github.com/aspnet/AspNetCore/pull/8374/commits/b0dca95ad87dab6adabf7f3665f94cdf5e10bdeb

Reply:
@davidfowl why the option? Is there any reason to turn this off other than to mitigate a bug in the implementation?

Reply:
From a runtime point of view the switch is faster since its a `readonly static` so branch eliminates one way or the other. The config gets evaluated on every iteration.

Reply:
No need for that, it only needs to be evaluated once.

Reply:
Wondering if Options can be "baked" to `readonly static`s currently
e.g.
```csharp
if (buffer.Length >= ServerOptions.Limits.MaxRequestLineSize)
```
is 3 indirections
```asm
G_M62071_IG06:
       488B9780000000       mov      rdx, gword ptr [rdi+128]
       488B5218             mov      rdx, gword ptr [rdx+24]
       8B5220               mov      edx, dword ptr [rdx+32]
       4C63CA               movsxd   r9, edx
       493BC9               cmp      rcx, r9
       7C37                 jl       SHORT G_M62072_IG07
```
if it can be "baked" then it would just be a const compare

```asm
G_M62071_IG06:
       493BC9               cmp      rcx, 8192
       7C37                 jl       SHORT G_M62072_IG07
```

Reply:
Probably not as it would cause a problem if you were running more than once Kestrel instance in a process :-/

Reply:
Squished the api commits; just in case want it reverted https://github.com/aspnet/AspNetCore/pull/8374/commits/b822a0fc1027cda4c39948dbb2b05e73067adfd3

Reply:
A little difference; but not huge

Config
```
               Method |     Mean |        Op/s | Scaled | Allocated |
--------------------- |---------:|------------:|-------:|----------:|
 PlaintextTechEmpower | 541.7 ns | 1,846,001.2 |   1.00 |       0 B |
           LiveAspNet | 891.1 ns | 1,122,221.7 |   1.65 |       0 B |
```         
Switch
```
               Method |     Mean |        Op/s | Scaled | Allocated |
--------------------- |---------:|------------:|-------:|----------:|
 PlaintextTechEmpower | 536.9 ns | 1,862,399.7 |   1.00 |       0 B |
           LiveAspNet | 878.1 ns | 1,138,780.0 |   1.64 |       0 B |
```